### ISSUE TITLE
Multiple issues with the climatic events dialogues

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#2836]
[R04_TIMEHRI_PICSA.txt](https://github.com/africanmathsinitiative/Instat/files/863664/R04_TIMEHRI_PICSA.txt)

[timehri.zip](https://github.com/africanmathsinitiative/Instat/files/863695/timehri.zip)

I attach the data file used yesterday for the start of the rains.  The zip file from Timehri already has the date column and day of year.  Please try the following with that file:

1) Go straight to your start of the rains.  
2) Note the dialogue partly completed automatically.  Hooray.
3) Leave the dates 1 to 366 as given.
4) Specify the amounts as 7 days, with the 0.80 percentile.  That's all.  Press OK
5)  It works.  Hooray again.
6) Rename the column generated - with the impossibly long name - to V5.  
7) **It gives an error**.  But then it seems to do it.
The error is as follows:
---------------------------
Error running R command(s)
---------------------------
Error: object 'new_value' not found
The error occurred in attempting to run the following R command(s):

InstatDataObject$rename_column_in_data(data_name="R04_TIMEHRI_PICSA_by_V1", column_name="First_Possible_Onset_wet_season", new_val="V5")

8) Repeat the start of the rains changing 7 days to 5 days.
9) Rename the column again, to V6.
10) Repeat the start of the rains again, changing 5 days back to 7 days.

Look at the results.  They are now very different to before!!!

I had various other problems running the start with your more complex definitions.  I assume things are not clearing from one run to another.  After a time it runs forever, without completing.

**Comment by rdstern** [comment]
By the way, I started with the other file, which has data from 3 stations.  That crashes immediately.  It needs data from only a single station.  I hope we will eventually be able to include data from multiple stations

**Comment by rdstern** [comment]
The climatic events are working much better now.  Well done indeed.

I used the data from Timehre - shared above I think, but could try with another.

Definition was from day 92 to 366.
7 day rainfall > 80% point.
4 days out of 7 with rain
It ran OK.
Then changed 4 days to 2 days.  Also ran OK.  But we noticed that the second set was sometimes later than the first set and this cannot be.  The condition is less stringent, so the start must be on the same day, or earlier.  I don't see it can be later.

As a small aside, when we get the percentile, it would be good to know  - in the output window what the corresponding amount was.
